File: java/lineal-structures/List.java
class ListNode {
    int data;
    ListNode next;
    public ListNode(int data) {
        this.data = data;
        this.next = null;
    }
}

public class List {
    ListNode head;

    void add(int val) {
        ListNode tmp =  new ListNode(val);
        tmp.next = head;
        head = tmp;
    }

    void show() {
        ListNode tmp = head;
        while(tmp!=null) {
            System.out.print(tmp.data+" ");
            tmp = tmp.next;
        }
    }
    
    public static void main(String [] args){
        List list = new List();
        list.add(5);
        list.add(10);
        list.show();
    }

}

File: go/lineal-structures/list.go
package main

import "fmt"

type Node struct {
	data int
	next *Node
}

type List struct {
	head *Node
}

func (l *List) add(data int) {
	tmp := &Node{data: data, next: l.head}
	l.head = tmp
}

func (l *List) show() {
	tmp := l.head
	for tmp != nil {
		fmt.Print(tmp.data, " ")
		tmp = tmp.next
	}
}

func main2() { //Go does not allow multiple main
	l := List{}
	l.add(3)
	l.add(9)
	l.add(6)
	l.show()
}
